### PR TITLE
.github: actually run tests in CI

### PR DESCRIPTION
**Pull request by maisem**
Updates #cleanup

### CODE REVIEW

**Review by bradfitz — APPROVED** [review]

File: util/deephash/deephash_test.go [hunk 1]
```diff
@@ -139,7 +139,7 @@ func TestHash(t *testing.T) {
 		{in: tuple{netip.AddrPortFrom(netip.AddrFrom4([4]byte{192, 168, 0, 1}), 1234), netip.AddrPortFrom(netip.AddrFrom4([4]byte{192, 168, 0, 1}), 1235)}, wantEq: false},
 		{in: tuple{netip.AddrPortFrom(netip.AddrFrom4([4]byte{192, 168, 0, 1}), 1234), netip.AddrPortFrom(netip.AddrFrom4([4]byte{192, 168, 0, 2}), 1234)}, wantEq: false},
 		{in: tuple{netip.Prefix{}, netip.Prefix{}}, wantEq: true},
-		{in: tuple{netip.Prefix{}, netip.PrefixFrom(netip.Addr{}, 1)}, wantEq: false},
+		{in: tuple{netip.Prefix{}, netip.PrefixFrom(netip.Addr{}, 1)}, wantEq: true},
```
Comment (maisem) [inline util/deephash/deephash_test.go:142]: New behavior in go1.21
cc @dsnet @bradfitz

Comment (maisem) [reply util/deephash/deephash_test.go:142]: https://cs.opensource.google/go/go/+/3693fd255f44acc24efa2b45317473d4d1e1abde

Comment (bradfitz) [reply util/deephash/deephash_test.go:142]: this needs to pass with both Go 1.20 and Go 1.21 for a few months. So maybe just remove this test case for now. It's not really material to the deephash package.

**Comment by maisem** [comment]
The regression was introduced in 43819309e18ac7d496643fcd86489a2ee0927964

**Review by bradfitz — APPROVED** [review]

File: util/deephash/deephash_test.go (hunk 1 above)
Comment (dsnet) [reply util/deephash/deephash_test.go]: Agreed with removing this case. I made the upstream change to have one consistent invalid representation.

**Pull request merged**